srtcore/core.cpp

```diff
@@ -9993,10 +9996,12 @@ int srt::CUDT::checkLazySpawnTsbPdThread()
 {
     const bool need_tsbpd = m_bTsbPd || m_bGroupTsbPd;
 
-    if (need_tsbpd && !m_RcvTsbPdThread.joinable())
-    {
-        ScopedLock lock(m_RcvTsbPdStartupLock);
+    if (!need_tsbpd)
+        return 0;
```

Comment:
This change has been extracted to PR #2972 as not dependent on the rest changes from that PR.

Reply:
No worries, if merged, it will disappear from here.